cylc/rose/platform_utils.py

```diff
@@ -47,15 +49,31 @@ def get_platform_from_task_def(
     config = WorkflowConfig(flow, flow_file, Values())
     # Get entire task spec to allow Cylc 7 platform from host guessing.
     task_spec = config.pcfg.get(['runtime', task])
+    # check for subshell and evaluate
+    if (
+        task_spec.get('platform', None) is not None and
+        is_platform_definition_subshell(task_spec['platform'])
+    ):
+        task_spec['platform'] = eval_subshell(task_spec['platform'])
+    elif (
+        task_spec.get('remote', {}).get('host', None) is not None and 
+        HOST_REC_COMMAND.match(task_spec['remote']['host'])
+    ):
+        task_spec['remote']['host'] = eval_subshell(
+            task_spec['remote']['host'])
     platform = get_platform(task_spec)
-    if platform is None:
-        raise PlatformLookupError(
-            'Platform lookup failed; platform is a subshell to be evaluated: '
-            f' Task: {task}, platform: {task_spec["platform"]}.'
-        )
```

Comment:
What's the deal with removing this exception?

(Sorry if I've missed the discussion on this)

Reply:
This is no longer going to be hit. Before the return was None if platform was a subshell. Now, the subshell evaluation is made before checking for platform so will no longer return None.

Reply:
Right, ok, it only returns `None` if it *is* a subshell, ta.